diff --git a/tools/hciattach_ti.c b/tools/hciattach_ti.c
index 8322b45..191f7cd 100644
--- a/tools/hciattach_ti.c
+++ b/tools/hciattach_ti.c
@@ -55,7 +55,7 @@
 
 #define TI_MANUFACTURER_ID	13
 
-#define FIRMWARE_DIRECTORY	"/lib/firmware/ti-connectivity/"
+#define FIRMWARE_DIRECTORY	"/legato/systems/current/apps/bluetoothUtil/read-only/bin/"
 
 #define ACTION_SEND_COMMAND	1
 #define ACTION_WAIT_EVENT	2
